crates/sui-rest-api/src/transactions/resolve/literal.rs

```diff
@@ -0,0 +1,787 @@
+        | Type::Reference(_)
+        | Type::MutableReference(_) => Err(RestError::new(
+            axum::http::StatusCode::BAD_REQUEST,
+            format!("literal cannot be resolved into type {type_}"),
+        )),
+    }
+}
+
+fn resolve_as_bool(buf: &mut Vec<u8>, value: &UnresolvedValue) -> Result<()> {
+    let b: bool = match value {
+        UnresolvedValue::Bool(b) => *b,
+        UnresolvedValue::String(s) => s.parse().map_err(|e| {
+            RestError::new(
+                axum::http::StatusCode::BAD_REQUEST,
+                format!("literal cannot be resolved as bool: {e}"),
+            )
+        })?,
+        UnresolvedValue::Null | UnresolvedValue::Number(_) | UnresolvedValue::Array(_) => {
+            return Err(RestError::new(
+                axum::http::StatusCode::BAD_REQUEST,
+                "literal cannot be resolved into type bool",
+            ))
+        }
+    };
+
+    bcs::serialize_into(buf, &b)?;
+
+    Ok(())
+}
+
+fn resolve_as_number<T>(buf: &mut Vec<u8>, value: &UnresolvedValue) -> Result<()>
+where
+    T: std::str::FromStr + TryFrom<u64> + serde::Serialize,
+    <T as std::str::FromStr>::Err: std::fmt::Display,
+    <T as TryFrom<u64>>::Error: std::fmt::Display,
+{
+    let n: T = match value {
+        UnresolvedValue::Number(n) => T::try_from(*n).map_err(|e| {
+            RestError::new(
+                axum::http::StatusCode::BAD_REQUEST,
+                format!(
+                    "literal cannot be resolved as {}: {e}",
+                    std::any::type_name::<T>()
+                ),
+            )
+        })?,
+
+        UnresolvedValue::String(s) => s.parse().map_err(|e| {
+            RestError::new(
+                axum::http::StatusCode::BAD_REQUEST,
+                format!(
+                    "literal cannot be resolved as {}: {e}",
+                    std::any::type_name::<T>()
+                ),
+            )
+        })?,
+
+        UnresolvedValue::Null | UnresolvedValue::Bool(_) | UnresolvedValue::Array(_) => {
+            return Err(RestError::new(
+                axum::http::StatusCode::BAD_REQUEST,
+                format!(
+                    "literal cannot be resolved into type {}",
+                    std::any::type_name::<T>()
+                ),
+            ))
+        }
+    };
+
+    bcs::serialize_into(buf, &n)?;
+
+    Ok(())
+}
+
+fn resolve_as_address(buf: &mut Vec<u8>, value: &UnresolvedValue) -> Result<()> {
+    let address = match value {
+        // parse as ObjectID to handle the case where 0x is present or missing
+        UnresolvedValue::String(s) => s.parse::<ObjectID>().map_err(|e| {
+            RestError::new(
+                axum::http::StatusCode::BAD_REQUEST,
+                format!("literal cannot be resolved as bool: {e}"),
+            )
+        })?,
+        UnresolvedValue::Null
+        | UnresolvedValue::Bool(_)
+        | UnresolvedValue::Number(_)
+        | UnresolvedValue::Array(_) => {
```

Comment:
```suggestion
       _
```

Reply:
I think this is a matter of preference, some times i like the blanket case other i like explicitly enumerating all the cases. Here it probably is fine to use the blanket case though

Reply:
Yep, there are ony 4 cases so it's probably fine. I did not scroll down to see the rest of this match in other places, so keeping it as is it's fine.